### PR TITLE
Use RAII for timing scopes instead of error-prone push-pop

### PR DESCRIPTION
**Pull request by dimhotepus**
Why?
1. It is easy to forget `pop()` call, so timing values may be missed.
2. Reduce src size, so the bugs probability.

Also fix small typo in CONTRIBUTING.md

### CODE REVIEW

**Comment by faberga** [comment]
Hi @dubek Can you review this PR, please?

**Comment by dimhotepus** [comment]
>    If you don't declare a variable name, the Guard object is constructed and immediately destructed (not at the end of the function). See the [ES.84 C++ guideline](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-noname) for explanations.

Thank you! That's easy to miss such thing with little experience in C++.

I also fixed small typo in CONTRIBUTING.md.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Adding new features, improving documentation, fixing bugs, writing new tests,
 designing and coding new examples or writing tutorials are all examples of 
 helpful contributions.
 
-## Prerequisits
+## Prerequisities
```
Comment (dubek) [inline CONTRIBUTING.md:8]: Thanks for the fix, but I think this is also the wrong spelling... See one line below.

Comment (dimhotepus) [reply CONTRIBUTING.md]: > Thanks for the fix, but I think this is also the wrong spelling... See one line below.

Right, changed to [prerequisites](https://www.merriam-webster.com/dictionary/prerequisites).

**Pull request merged**